java/source/dataset.rst

```diff
@@ -0,0 +1,232 @@
+.. _arrow-dataset:
+
+=======
+Dataset
+=======
+
+Current java project that use JNI are:
+
+* `Arrow Java Dataset <https://arrow.apache.org/docs/dev/java/dataset.html>`_: Java implementation of Arrow Dataset API/Framework. JniLoader [arrow_cdata_jni]
+* `Arrow Java C Data Interface <https://arrow.apache.org/docs/format/CDataInterface.html>`_: Java implementation of C Data Interface. JniLoader [arrow_dataset_jni]
```

Comment:
Sorry, what are the `JniLoader [arrow_cdata_jni]` snippets here referring to? (Also, they seem to be swapped between the two?)

Also, it's just called the `Arrow C Data Interface`. However, given this file is named "Dataset", maybe let's not even talk about the C Data Interface here - that can be a separate file.

Reply:
Changed